### ISSUE TITLE
Flutter 3.7 breaks a lot of things

### ISSUE THREAD

**Issue by suragch** [suragch/mongol#38]
The Flutter 3.7 update introduced a lot of changes to the text editing classes and also some changes to the text widgets. I'm trying to update the related classes now in the `mongol` library but it is taking more work than I originally expected.

**Comment by suragch** [comment]
Updating on this branch: https://github.com/suragch/mongol/tree/flutter3-7

**Comment by suragch** [comment]
The library is running now but text editing has several problems. I'll document these later.

**Comment by suragch** [comment]
@Satsrag Have you upgraded to Flutter 3.7 yet?

**Comment by suragch** [comment]
It seems that the errors are related to the input decoration. I can't find any more problems in `MongolEditableText` layer and lower.

**Comment by Satsrag** [comment]
I tested about `TextInputControl` and `TextFieldTapRegion`. They were working well. I added the  [new keyboard demo](https://github.com/Satsrag/mongol/commit/cc57d549bddc33d595c44ae127871a9eb9954af1) using `TextInputControl` for testing. If it is applicable here, I will `pull request`.
I think #25 should be closed because `TextFieldTapRegion` is the right solution for it.

**Comment by suragch** [comment]
@Satsrag Thank you for checking. I've published version 4.0.0 now. `MongolTextField` still has the following issues: #29, #30, #32, #39, #40, #41. However, I think things are working enough so that we can get something out for people who are upgrading to Flutter 3.7. I'll keep trying to fix those issues, but if you find a solution too, that would be great. I broke something about the selection handles and context menu when I copied the Flutter 3.7 code over.

That's good to know about the `TextFieldTapRegion`. Yes, if you make a PR for your new keyboard demo, that would be great. I'll close #25 after that.
